drivers/battery/Kconfig

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig BATTERY
+	bool "Battery drivers"
+	help
+	  Enable battery driver configuration.
+
+if BATTERY
+
+config BATTERY_SHIP_MODE_WRITES
+	int "Number of writes needed to invoke battery cutoff command"
+	default 2
+	help
+	  Ship mode electronically disconnects the battery from the rest of the
+	  device to minimize power drain while the device is idle.
+
+config BATTERY_LEVEL_FULL
+	int "Battery-full level percentage"
+	default 100
+	help
+	  Stop charge when charging and battery level >= this percentage.
+
+config BATTERY_LEVEL_LOW
+	int "Battery-critical level percentage."
+	default 5
+	help
+	  Send battery-critical event when discharging and battery level <= this
+	  level.
+
+config BATTERY_LEVEL_SHUTDOWN
+	int "Device shutdown level percentage."
+	default 3
+	help
+	  Shut down device when discharging and battery level < this
```

Comment:
Is it obvious to other devs how 'shutdown' would happen? I'm thinking there are two sensible options - send an event, or call a system shutdown function. Perhaps a third - call a custom function. Does this need an alert-mode config?